### PR TITLE
Luke Bindon 46992505 2D Unet pull request

### PR DESCRIPTION
**Pull request by lukeybin**
### Summary
This pull request adds the implementation of a U-Net model for medical image segmentation, including data augmentation and optimisation techniques (Dice loss and Dice coefficient).

### Changes
- Added `unet_model` with custom Dice loss and metric.
- Implemented data augmentation using `ImageDataGenerator`.

### Testing
The model was tested on the training and validation datasets. Plots for loss and accuracy were generated.

### Feedback
I am keen to receive and act upon any and all feedback on my model implementation and architecture.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.71. 

**Model and functions:**

- It correctly uses Tensorflow to construct the 2D-UNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Minimal function docstrings

**Difficulty:** Easy.

**Additional Comments:**

- Reasonable commits
- The report is expected to contain the user guide, conclusion, and reference.
- The figure to assess model performance should be the Dice Similarity Coefficient or Dice score instead of accuracy.
- The loss figure indicates an overfitting occurred.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Some overfitting.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Minimal doc strings.</td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF NOT submitted. PDF NOT submitted. </td><td>-2</td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-15</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by gayanku** [comment]
Pls submit PDF.

**Comment by lukeybin** [comment]
Sorry, am I still able to submit? Should I send to you directly
